packages/react/src/components/Comments/Thread.tsx

```diff
@@ -1,53 +1,20 @@
 import { mergeCSSClasses } from "@blocknote/core";
-import { useCallback, useMemo } from "react";
+import { useCallback, useMemo, useState } from "react";
 import { useComponentsContext } from "../../editor/ComponentsContext.js";
 import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
 import { useCreateBlockNote } from "../../hooks/useCreateBlockNote.js";
 import { useDictionary } from "../../i18n/dictionary.js";
-import { Comment, CommentProps } from "./Comment.js";
+import { Comment } from "./Comment.js";
 import { CommentEditor } from "./CommentEditor.js";
 import { schema } from "./schema.js";
-import { useThreads } from "./useThreads.js";
 import { useUsers } from "./useUsers.js";
+import { ThreadData } from "@blocknote/core/comments";
 
 export type ThreadProps = {
-  /**
-   * The thread to display.
-   */
-  threadId: string;
-
-  /**
-   * Whether to show the composer to reply to the thread.
-   */
-  showComposer?: boolean;
-
-  /**
-   * Whether to show the action to resolve the thread.
-   */
-  showResolveAction?: boolean;
-
-  /**
-   * How to show or hide the actions.
-   */
-  showActions?: CommentProps["showActions"];
-
-  /**
-   * Whether to show reactions.
-   */
-  showReactions?: CommentProps["showReactions"];
-
-  /**
-   * Whether to show deleted comments.
-   */
-  showDeletedComments?: CommentProps["showDeleted"];
-
-  /**
-   * Component element props.
-   */
-  className?: string;
-  onFocus?: () => void;
-  onBlur?: () => void;
-  tabIndex?: number;
+  thread: ThreadData;
+  view: "floating" | "sidebar";
```

Comment:
same for `index` earlier. Do you prefer this component to have a `view` prop, or the props we had before?

Reply:
Here I think using `view` is nicer - `onFocus`, `onBlur`, and `tabIndex` can all be consolidated into a single prop. Also it's much clearer what `view` is for, whereas the props it replaces are all generic don't immediately tell you why that their point is.